packages/nimble-components/src/table/specs/table-column-width-hld.md

```diff
@@ -0,0 +1,189 @@
+
+### Out of Scope
+
+There are some column sizing behaviors that we will ultimately expect to support, but the APIs presented here are not meant to address:
+
+-   Auto-resizing: We will not describe how we intend to support the use-case of having a column auto-size to its contents
+-   Different interactive sizing modes: While the APIs described in this HLD do not inherently prescribe to a particular interactive sizing behavior, it's worth saying that in order to support multiple sizing modes, there will likely be additional APIs required that this HLD does not address.
+-   Mechanisms related to accessibility-centric interactive column sizing (if there are such mechanisms). One possible example is allowing a user to size a column by way of the keyboard, instead of using a mouse. Ultimately, such a scenario is not in conflict with the API presented here, nor the mouse-based approach we know we will require, and can thus be handled separately, if ever.
+
+### API
+
+To accomodate the various sizing modes of a column, in addition to the other necessary behaviors, we can add the following properties to `TableColumn`:
+
+```
+abstract class TableColumn {
+    /*
+     * @internal
+     */
+    // when set 'currentFractionalWidth' will be ignored
+    @observable
+    public currentPixelWidth: number | null = null;
+
+    /*
+     * @internal
+     */
+    @observable
+    public currentFractionalWidth = 1;
+
+    /*
+     * @internal
+     */
+    @observable
+    public columnMinWidth = 88; // the minimum size in pixels according to the design doc
+
+    /*
+     * @internal
+     */
+    @observable
+    public currentDisableResize = false;
+
+    ...
+}
+```
+
+Note that these properties are not attributes, and thus are not set by clients on the components via markup. Instead, these properties are meant to be configured by the concrete column implementations (typically initialization), as well as the Table as part of interactive column sizing.
+
+#### `currentFractionalWidth` vs `currentPixelWidth` behavior
+
+-   The values supplied to the `currentFractionalWidth` property have the same meaning as the values supplied with the [`fr` unit](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Grid_Layout/Basic_Concepts_of_Grid_Layout#the_fr_unit) for the CSS `grid-template-columns` property.
+-   The values supplied to the `currentPixelWidth` property are meant to be pixel-based values.
+-   The table will use `currentPixelWidth` over `currentFractionalWidth` when both are set.
+
+#### `columnMinWidth` behavior
+
+Open Question: Is there a minimum width considered too small? If a plugin is allowed to provide a mimimum width smaller than the default, how should the various components within the header behave? If a plugin attempts to set a minimum width to an unsupported value, what is the appropriate response (i.e. should we simply update the `TableValidity` state?)?
```

Comment:
If this is an open issue we don't need to address in the HLD maybe it should move to the Open Issues section at the bottom (makes it easier to see the HLD open Qs).

I think a couple of these may be resolved from our quick convo, ie currentMinWidth / columnMinPixelWidth just represents the min width of a cell which is independent from the header which will define it's own min width (based on min width of label text, icon presenece, sort arrow presence, etc.). So the plugin can report whatever it wants (ie as low as zero).

Reply:
Yeah, I've just removed this section.